src/initialize.cc

```diff
@@ -4,13 +4,28 @@
  * \brief initialize mxnet library
  */
 #include <dmlc/logging.h>
+#include <mxnet/engine.h>
+
+#include "engine/profiler.h"
 
 namespace mxnet {
 
 class LibraryInitializer {
  public:
   LibraryInitializer() {
     dmlc::InitLogging("mxnet");
+#if MXNET_USE_PROFILER
+    // ensure engine's and profiler's constructor are called before atexit.
+    Engine::Get();
+    engine::Profiler::Get();
```

Comment:
the LibraryInitializer call in dllmain with windows, it will deadlock.please move it to other.